// @ionic-native/aes-256/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var AES256 = /** @class */ (function (_super) {
    __extends(AES256, _super);
    function AES256() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AES256.prototype.encrypt = function (secureKey, secureIV, data) { return cordova(this, "encrypt", {}, arguments); };
    AES256.prototype.decrypt = function (secureKey, secureIV, data) { return cordova(this, "decrypt", {}, arguments); };
    AES256.prototype.generateSecureKey = function (password) { return cordova(this, "generateSecureKey", {}, arguments); };
    AES256.prototype.generateSecureIV = function (password) { return cordova(this, "generateSecureIV", {}, arguments); };
    AES256.pluginName = "AES256";
    AES256.plugin = "cordova-plugin-aes256-encryption";
    AES256.pluginRef = "cordova.plugins.AES256";
    AES256.repo = "https://github.com/Ideas2IT/cordova-aes256";
    AES256.platforms = ["Android", "iOS"];
    AES256.install = "ionic cordova plugin add cordova-plugin-aes256-encryption";
AES256.ɵfac = function AES256_Factory(t) { return ɵAES256_BaseFactory(t || AES256); };
AES256.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: AES256, factory: function (t) { return AES256.ɵfac(t); } });
var ɵAES256_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(AES256);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AES256, [{
        type: Injectable
    }], null, null); })();
    return AES256;
}(IonicNativePlugin));
export { AES256 };
